test/modellib/ModelLib.hpp

```diff
@@ -97,3 +99,67 @@ bool genLSTMModelAndCompile(
     llvm::SmallVector<int64_t, 3> &hShape,
     llvm::SmallVector<int64_t, 3> &cShape, OMTensor *&wOmt, OMTensor *&rOmt,
     OMTensor *&bOmt, OMTensor *&pOmt);
+
+class ModelLibBuilder {
+public:
+  // Define the model. Subclass should add to the builder all of the specific
+  // parameters that uniquely define the model.
+  ModelLibBuilder(const std::string &sharedLibBaseName);
+  // Destructor needed to free the inputs/outputs data structures.
+  ~ModelLibBuilder();
```

Comment:
This class acts as an abstract class defining an interface that derived classes must implement. The destructor should be made virtual, that way if a `ModelLibBuilder*` is deleted, and the pointer points to a derived class the destructor of the derived class will be run, giving the derived class a chance to clean up resources (if needed).

Reply:
ok, fully addressed by adding virtual